main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -70,72 +59,47 @@
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    Map<String, GtfsAgency> agencyByUrlMap = agenciesByUrlMap(agencyTable);
+    ListMultimap<String, GtfsAgency> agencyByUrlMap = agenciesByUrlMap(agencyTable);
+    ListMultimap<String, GtfsRoute> routesByUrlMap = routesByUrlMap(routeTable);
     for (GtfsRoute route : routeTable.getEntities()) {
       if (!route.hasRouteUrl()) {
-        return;
+        continue;
       }
-      GtfsAgency agency = agencyByUrlMap.get(route.routeUrl().toLowerCase());
-      if (agency != null) {
-        noticeContainer.addValidationNotice(
-            new SameRouteAndAgencyUrlNotice(
-                route.csvRowNumber(),
-                route.routeId(),
-                agency.agencyName(),
-                route.routeUrl(),
-                agency.csvRowNumber()));
+      List<GtfsAgency> agencies = agencyByUrlMap.get(route.routeUrl().toLowerCase());
+      for (GtfsAgency agency : agencies) {
```

Comment:
I like your idea to emit notices for all pairs of objects.

Reply:
I'm glad you like it :)